Tests/ParseSwiftTests/ParseUserTests.swift

```diff
@@ -1083,59 +1083,75 @@ class ParseUserTests: XCTestCase { // swiftlint:disable:this type_body_length
-            }
-
-            if let installationFromKeychain = BaseParseInstallation.current {
-                XCTFail("\(installationFromKeychain) wasn't deleted from Keychain during logout")
+                XCTFail(error.localizedDescription)
             }
-        } catch {
-            XCTFail(error.localizedDescription)
         }
     }
 
     func logoutAsync(callbackQueue: DispatchQueue) {
 
         let expectation1 = XCTestExpectation(description: "Logout user1")
-        User.logout(callbackQueue: callbackQueue) { result in
 
-            switch result {
+        DispatchQueue.main.async {
+            let oldInstallationId = BaseParseInstallation.current?.installationId
+            User.logout(callbackQueue: callbackQueue) { result in
 
-            case .success:
-                if let userFromKeychain = BaseParseUser.current {
-                    XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
-                }
+                switch result {
 
-                if let installationFromMemory: CurrentInstallationContainer<BaseParseInstallation>
-                    = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
-                    XCTFail("\(installationFromMemory) wasn't deleted from memory during logout")
-                }
+                case .success:
+                    if let userFromKeychain = BaseParseUser.current {
+                        XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
+                    }
+                    DispatchQueue.main.async {
+                        if let installationFromMemory: CurrentInstallationContainer<BaseParseInstallation>
+                            = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
+                            if installationFromMemory.installationId == oldInstallationId {
+                                // swiftlint:disable:next line_length
+                                XCTFail("\(installationFromMemory) wasn't deleted and recreated in memory during logout")
+                            }
+                        }
 
-                #if !os(Linux) && !os(Android)
-                if let installationFromKeychain: CurrentInstallationContainer<BaseParseInstallation>
-                    = try? KeychainStore.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
-                    XCTFail("\(installationFromKeychain) wasn't deleted from Keychain during logout")
-                }
-                #endif
+                        #if !os(Linux) && !os(Android)
+                        if let installationFromKeychain: CurrentInstallationContainer<BaseParseInstallation>
+                            = try? KeychainStore.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
+                            if installationFromKeychain.installationId == oldInstallationId {
```

Comment:
What do you think about extending the test case here and also checking if `installationFromKeychain.currentInstallation != nil`?

Reply:
This may not always be the case by the time the test gets here because of the dispatching to main queue. If it is nil when it gets here, it won't be in a couple of seconds

Reply:
You're right. I haven't thought about the dispatch. The additional test case is not required. Has just been a thought as it's not checked right now.

Reply:
#113 should provide more predictable behavior. Let me know your thoughts on that thread